src/pages/Models/Details/generators.js

```diff
@@ -0,0 +1,318 @@
+import React from "react";
+import classnames from "classnames";
+
+import {
+  generateStatusElement,
+  generateIconPath,
+  generateSpanClass
+} from "app/utils";
+
+export const applicationTableHeaders = [
+  { content: "app" },
+  { content: "status" },
+  { content: "version", className: "u-align--right" },
+  { content: "scale", className: "u-align--right" },
+  { content: "store" },
+  { content: "rev", className: "u-align--right" },
+  { content: "os" },
+  { content: "notes" }
+];
+
+export const unitTableHeaders = [
+  { content: "unit" },
+  { content: "workload" },
+  { content: "agent" },
+  { content: "machine", className: "u-align--right" },
+  { content: "public address" },
+  { content: "port", className: "u-align--right" },
+  { content: "message" }
+];
+
+export const machineTableHeaders = [
+  { content: "machine" },
+  { content: "state" },
+  { content: "az" },
+  { content: "instance id" },
+  { content: "message" }
+];
+
+export const relationTableHeaders = [
+  { content: "relation provider" },
+  { content: "requirer" },
+  { content: "interface" },
+  { content: "type" },
+  { content: "message" }
+];
+
+export function generateIconImg(name, namespace) {
+  return (
+    <img
+      alt={name + " icon"}
+      width="24"
+      height="24"
+      className="entity-icon"
+      src={generateIconPath(namespace)}
+    />
+  );
+}
+
+export function generateEntityLink(namespace, href, name, subordinate) {
+  return (
+    <>
+      {subordinate && <span className="subordinate"></span>}
+      {namespace && generateIconImg(name, namespace)}
+      <a href={href}>{name}</a>
+    </>
+  );
+}
+
+export function generateApplicationRows(
+  modelStatusData,
+  filterByApp,
+  onRowClick
+) {
+  if (!modelStatusData) {
+    return [];
+  }
+
+  const applications = modelStatusData.applications;
+
+  return Object.keys(applications).map(key => {
+    const app = applications[key];
+    return {
+      columns: [
+        {
+          content: generateEntityLink(app.charm || "", "#", key),
+          className: "u-display--flex"
+        },
+        {
+          content: app.status ? generateStatusElement(app.status.status) : "-",
+          className: "u-capitalise"
+        },
+        { content: "-", className: "u-align--right" },
+        { content: "-", className: "u-align--right" },
+        { content: "CharmHub" },
+        { content: key.split("-")[-1] || "-", className: "u-align--right" },
+        { content: "Ubuntu" },
+        { content: "-" }
+      ],
+      className: filterByApp === key ? "is-selected" : "",
```

Comment:
These functions were all copied over from the other file with the exception of these few lines on each table generation to add the necessary data for the filtering selection.